CONTRIBUTING.md

```diff
@@ -0,0 +1,49 @@
+# Contributing Guidelines
+
+This document contains information and guidelines about contributing to this project. Please read it before you start participating.
+
+**Topics**
+
+- [Asking Questions](#asking-questions)
+- [Reporting Issues](#reporting-issues)
+- [Project Setup](#project-setup)
+
+## Asking Questions
+
+We don't use GitHub as a support forum. For any usage questions that are not specific to the project itself, please ask on [Stack Overflow](https://stackoverflow.com) instead. By doing so, you'll be more likely to quickly solve your problem, and you'll allow anyone else with the same question to find the answer. This also allows maintainers to focus on improving the project for others.
+
+## Reporting Issues
+
+A great way to contribute to the project is to send a detailed issue when you encounter a problem. We always appreciate a well-written, thorough bug reports.
+
+When reporting issues, please include the following:
+
+- The version of Xcode you're using
+- The version of iOS or macOS you're targeting
+- The full output of any stack trace or compiler error
+- A code snippet that reproduces the described behavior, if applicable
+- Any other details that would be useful in understanding the problem
+
+This information will help us review and fix your issue faster.
+
+## Project Setup
+
+Before you can start developing a fix or a new feature in this project you need to be aware of how this project is structured. Especially since SwiftGen is a command line tool written in Swift some things need to be handled differently.
+
+### Requirements
+
+First you need to make sure your machine fulfills the following requirements:
+
+- You have [Homebrew](http://brew.sh), [CocoaPods](https://cocoapods.org) and [xcpretty](https://github.com/supermarin/xcpretty) installed
```

Comment:
I agree on all points. Btw (and this is a bit off topic) when working on the dot syntax feature today I saw some code that I'm used to get warnings for from SwiftLint within the project like the `[String:AnyObject]` [here](https://github.com/AliSoftware/SwiftGen/blob/master/GenumKit/Stencil/Contexts.swift#L129). Just mentioning it, we could also create an issue for that if there isn't one already. I'm keeping those lines as is for now.